File: Linear_search/Dart/linear_search.dart
int? LinearSearch(List<int> arr, int target) {
    for (int i = 0; i < arr.length; i++) {
        if (arr[i] == target) {
            return i;
        }
    }
    return null;
}


void main() {
    List<int> arr = [5412, 6512, 777, 1];
    int target = 1;

    int? index = LinearSearch(arr, target);
    if (index != null) {
        print('The element $target was found at position ${index + 1}');
    } else {
        print('The element $target was not found');
    }
}

File: Linear_search/Zig/linear_search.zig
const std = @import("std");

pub fn main() !void {
  const arr = [_]i32{421, 62143, 1, 64};
  const target : i32 = 1;
  const index = LinearSearch(&arr, target);

  if (index) |pos|{
      std.debug.print("The element {d} was found at position {d}\n", .{ target, pos + 1});
  } else {
      std.debug.print("The element {d} was not found\n", .{target});
  }
}

fn LinearSearch(arr: []const i32, target: i32) ?usize {
    for (arr,0..) |value, i| {
        if (value == target) {
            return i;
        }
    }
    return null;
}
